students/Pirouz_N/lesson03/slicinglab.py

```diff
@@ -0,0 +1,502 @@
+"""
+Purpose: Lessen 3 homework one python certificate from UW
+Author: Pirouz Naghavi
+Date: 06/26/2020
+"""
+
+# Imports
+import copy
+from collections import Sequence
+import unittest
+
+
+def check_sequence(seq, minlength):
+    """"This function checks to see if the sequence is a sequence and has the right length.
+
+        Args:
+            seq: Is the sequence passed to the function to check.
+            minlength: Is the minimum length required.
+
+        Raises:
+            ValueError: If length of sequence is less than minlength.
+            TypeError: If None is passed instead of sequence.
+                If seq is not of type collection.Sequence.
+        """
+
+    if seq is None:
+        raise TypeError('Sequence is None.')
+    if not isinstance(seq, Sequence):
+        raise TypeError('Sequence is not a member of collections.Sequence.')
+    if len(seq) < minlength:
+        raise ValueError('Minimum length of sequence is {}.'.format(minlength))
+
+
+def exchange_first_last(seq):
+    """"This function swaps first element and last element of list.
+
+    Args:
+        seq: Is the sequence passed to the function to swap first and last element of.
+
+    Returns:
+        A shallow copy of sequence where the first and last elements are exchanged.
+
+    Raises:
+        ValueError: If length of sequence is less than one.
+        TypeError: If None is passed instead of sequence.
+            If seq is not of type collection.Sequence.
+    """
+    # Protecting against bad inputs
+    check_sequence(seq, 1)
+
+    # Creating a shallow copy of the sequence.
+    # new_seq = lis(seq) would work the same way as a shallow copy
+    new_seq = seq[:]
+
+    # Sequence length one needs to be handled separately
+    if len(new_seq) == 1:
+        return new_seq
+
+    # Creating memory placeholders for for first and last elements
+    first = new_seq[0: 1]
+    last = new_seq[len(seq)-1: len(seq)]
+
+    # Returning corrected sequence
+    return last + new_seq[1: len(new_seq) - 1] + first
```

Comment:
this can be simplified to `return seq[-1:] + seq[1:-1] + seq[:1]`

Reply:
Corrected it! 
Great point I wasn't entirely comfortable with using index -1, but getting more comfortable. Thank you.

Best Regards
Pirouz Naghavi